tmt/result.py

```diff
@@ -125,6 +125,11 @@ def show(self) -> str:
         return ' '.join(components)
 
 
+@dataclasses.dataclass
+class PhaseResult(BaseResult):
+    """ Describes what tmt knows about a single test phase result (sub-result) """
+
```

Comment:
A phase result should include also `check` field, see the specification at https://tmt.readthedocs.io/en/stable/spec/plans.html#results-format - AFAICT, it’s the only extra field beyond the base result.

Reply:
So... we will store check results for individual phases, right, makes sense, good point!

Reply:
@happz thanks for the catch. Hopefully, it should be resolved now.

Reply:
Correct, per-phase check results are one of the requirements, the one that forces us into reworking checks to be able to materialize as scripts on guests between phases.